resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/LocalDirsFeatureStep.scala

```diff
@@ -43,33 +44,52 @@ private[spark] class LocalDirsFeatureStep(
     val localDirVolumes = resolvedLocalDirs
       .zipWithIndex
       .map { case (localDir, index) =>
-        new VolumeBuilder()
-          .withName(s"spark-local-dir-${index + 1}")
-          .withNewEmptyDir()
-            .withMedium(if (useLocalDirTmpFs) "Memory" else null)
-          .endEmptyDir()
-          .build()
+        val name = s"spark-local-dir-${index + 1}"
+        findVolume(pod, name) match {
+          case Some(volume) => volume
```

Comment:
integration tests should exercise both cases (found the volume or created it)

Reply:
I agree with you, there is an existing unit test for creating a local directory via emptyDir, this PR adds a unit test for the case of found volume. Both cases should already be covered. Is that ok?